enhancements/cloud-integration/openstack/cloud-provider-config-upgrade.md

```diff
@@ -0,0 +1,327 @@
+should require no additional user input.
+
+#### Story 3 - Upgrade with non-default compatible configuration
+
+The Acme Corporation wants to upgrade their OCP 4.10 cluster to OCP 4.11.
+Following the user documentation [here][cloud-provider-options], they have
+configured a specific floating network ID. Upon upgrading, their cluster should
+automatically use CCM following the upgrade, and the upgrade should require no
+user input for this aspect of the upgrade. After the upgrade, their cluster
+should continue to use the same floating network ID.
+
+[cloud-provider-options]: https://docs.openshift.com/container-platform/4.9/installing/installing_openstack/installing-openstack-installer-custom.html#installation-osp-setting-cloud-provider-options_installing-openstack-installer-custom
+
+#### Story 4 - Upgrade with non-default incompatible configuration
+
+Omni Consumer Products wants to upgrade their OCP 4.10 cluster to OCP 4.11.
+They have configured a separate secret containing their cloud credentials as
+well as additional config directives. Upon attempting to upgrade, they discover
+they cannot as the Cluster Config Operator is marked as non-upgradable. The
+status of this operator informs them of the incompatible config directives and
+provides guidance on how to remedy the situation.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigations
+
+**Risk** Identical config options could work differently between the legacy and
+external cloud providers, or the conversion could be lossy.
+
+**Mitigation** Auditing and testing of the configuration options and
+combinations of same. At a minimum, we will explicitly test every legacy config
+option listed in the user documentation.
+
+**Risk** There may be additional configuration required for CCM compared to the
+legacy cloud provider in certain cloud configurations.
+
+**Mitigation** We currently test multiple cloud configurations. We should test
+an upgraded cluster on each of these cloud configurations.
+
+## Design Details
+
+### Configuration transformation for OpenStack CCM
+
+The configuration of OpenStack CCM is a superset of the legacy cloud provider
+configuration, with four exceptions. Details on how we will address each of
+these are provided below.
+
+<dl>
+<dt>secret-name</dt>
+<dt>secret-namespace</dt>
+<dt>kubeconfig-path</dt>
+<dd>
+  These are no longer relevant or supported in CCM. They define the cloud
+  credentials that are used by the legacy cloud provider. CCO will automatically
+  replace these with static config pointing to the cluster-managed cloud
+  credentials.
+  <br>
+  It is possible that the secret contains more information that the
+  cloud credentials populated by the installer. If so, the updated
+  configuration will no longer include these manual edits. This is
+  acceptable since they've done a dirty thing and deserve pain.
```

Comment:
> This is acceptable since they've done a dirty thing and deserve pain.

I'm not sure that's an acceptable support stance 😅  Can you think of an example of why a customer might have done manual edits here? Do we have a workaround for this supposed scenario?

Reply:
I think saying "additional information within the secret is not supported as we do not test or document it" or words to that effect is an acceptable way to finish this paragraph off, that alleviates my concerns at least